packages/ui5-library-generator/src/app/index.ts

```diff
@@ -0,0 +1,118 @@
+import Generator from 'yeoman-generator';
+import { AppWizard, Prompts } from '@sap-devx/yeoman-ui-types';
+import { join } from 'path';
+import ReuseLibGenLogger from '../utils/logger';
+import { t, prompts, initialAnswers, runPostLibGenHook, generatorTitle } from '../utils';
+import { platform } from 'os';
+import { CommandRunner } from '@sap-ux/nodejs-utils';
+import { ToolsLogger } from '@sap-ux/logger';
+import {
+    isCli,
+    getDefaultTargetFolder,
+    type YeomanEnvironment,
+    type VSCodeInstance
+} from '@sap-ux/fiori-generator-shared';
+import { generate, type UI5LibConfig } from '@sap-ux/ui5-library-writer';
+import { prompt, type UI5LibraryAnswers, type InquirerAdapter } from '@sap-ux/ui5-library-inquirer';
+import type { Ui5LibGenerator } from './types';
+
+/**
+ * Generator for creating a new UI5 library.
+ *
+ * @extends Generator
+ */
+export default class extends Generator implements Ui5LibGenerator {
+    answers: UI5LibraryAnswers = initialAnswers;
+    prompts: Prompts;
+    appWizard: AppWizard;
+    targetFolder: string;
+    vscode: unknown;
+    projectPath: string;
+    setPromptsCallback: (fn: any) => void;
+
+    /**
+     * Constructor for the generator.
+     *
+     * @param args - arguments passed to the generator
+     * @param opts - options passed to the generator
+     */
+    constructor(args: string | string[], opts: Generator.GeneratorOptions) {
+        super(args, opts);
+
+        this.appWizard = AppWizard.create(opts);
+        this.vscode = opts.vscode;
+        ReuseLibGenLogger.logger = new ToolsLogger({ logPrefix: '@sap-ux/generator-ui5-library' });
```

Comment:
This is no longer integrated to YUI logging. How will this work now when integrated back into YUI, it needs to allow the YUI logger to also be used and only create a new logger when one is not configured via YUI settings.

Reply:
Added logWrapper in fiori gen shared, as I expect it will be needed going forward